pkg/dashboard/server/alertStateManager.go

```diff
@@ -12,35 +12,67 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-type podStateManager struct {
+type alertStateManager struct {
 	notifManager notifications.Manager
 	store        store.Store
 	waitTime     time.Duration
 }
 
-func NewPodStateManager(notifManager notifications.Manager, store store.Store, waitTime time.Duration) *podStateManager {
-	return &podStateManager{notifManager: notifManager, store: store, waitTime: waitTime}
+func NewAlertStateManager(notifManager notifications.Manager, store store.Store, waitTime time.Duration) *alertStateManager {
+	return &alertStateManager{notifManager: notifManager, store: store, waitTime: waitTime}
 }
 
-func (p podStateManager) Track(pods []*api.Pod) {
+func (p alertStateManager) Track(pods []*api.Pod) {
 	p.trackStates(pods)
 }
 
-func (p podStateManager) Run() {
+func (p alertStateManager) TrackAlerts(alerts []api.Alert) {
```

Comment:
I think you are going to track raw K8s data:
- events
- and pod states

You don't necessarily need to abstract the input. But you have to abstract the alert itself.